mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp

```diff
@@ -87,8 +103,25 @@ void GPUToSPIRVPass::runOnOperation() {
   // Run conversion for each module independently as they can have different
   // TargetEnv attributes.
   for (Operation *gpuModule : gpuModules) {
-    spirv::TargetEnvAttr targetAttr =
-        spirv::lookupTargetEnvOrDefault(gpuModule);
+    auto constraints = cast<gpu::GPUModuleOp>(gpuModule).getConstraints();
+    auto reqdSubgroupSize =
+        constraints ? constraints->getReqdSubgroupSize() : std::nullopt;
+
+    spirv::TargetEnvAttr targetAttr = spirv::lookupTargetEnv(gpuModule);
+    const bool preexistingTargetEnv = targetAttr != nullptr;
+    assert(
+        !(preexistingTargetEnv && reqdSubgroupSize.has_value()) ||
+        static_cast<unsigned>(
+            targetAttr.getResourceLimits().getSubgroupSize()) ==
+                reqdSubgroupSize.value() &&
+            "SRIR-V target environment does not match gpu.reqd_subgroup_size");
+    if (!preexistingTargetEnv) {
+      targetAttr = spirv::getDefaultTargetEnv(&getContext());
+      if (reqdSubgroupSize) {
+        targetAttr =
+            spirvTargetWithSubgroupSize(targetAttr, reqdSubgroupSize.value());
+      }
+    }
```

Comment:
I think we are missing to update the other properties.

Reply:
I've added maxWorkgroupSize too, but reqdWorkgroupSize wasn't supported. What do you think?

Reply:
I'd say just setting min and max workgroup size to the same value should encode this

Reply:
Facts